build_tools/benchmarks/post_benchmark_comment_test.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from unittest import mock
+import http.client
+import requests
+import unittest
+
+import post_benchmark_comment
+
+
+class PostBenchmarkCommentTest(unittest.TestCase):
+
+  def test_post_to_gist(self):
+    gist_url = "https://example.com/123455/1234.md"
+    response = mock.create_autospec(requests.Response)
+    response.status_code = http.client.CREATED
+    response.json.return_value = {"html_url": gist_url, "truncated": False}
+    requester = mock.create_autospec(post_benchmark_comment.APIRequester)
```

Comment:
It's a bit odd to mock a class from the code under test here, though I think that class is so simple that it doesn't really need tests of its own. How about putting the creation and teardown of the mocks in a `setUp` method? That also ensures that the test doesn't accidentally make network calls.

https://docs.python.org/3/library/unittest.mock.html#patch-methods-start-and-stop

```python
    def setUp(self):
        self.mock_response = mock.create_autospec(requests.Response).start()
        self.mock_requester = mock.create_autospec(post_benchmark_comment.APIRequester).start()
        self.addCleanup(mock.patch.stopall())

```

Reply:
Since we make the `APIRequester` as an explicit parameter in these methods, there is no need to patch the class.

The mock here is used to create a fake `APIRequester` so we can pass it into the method under test and observe the interaction.

I think this is clearer than patching the classes used internally by the method under test. But I simply bring this pattern from Java best practices, so I might be wrong here : )

Reply:
Found some best practices (e.g. http://mauveweb.co.uk/posts/2014/09/every-mock-patch-is-a-little-smell.html) also mentioned that using DI + mock is better than patching

Reply:
Yeah I think the difference is just that usually you'd have this other class in a separate file and it would have its own unit tests. I can't really imagine a useful unit test for the class though, since it's so straightforward and in Python we don't have Java limitations about every file is a class. I agree that injecting the mock here is better than mocking out implementation details. What I meant to suggest was that you create the mock in the setup method so you don't have to do it every time. Perhaps what I suggested would actually mock all creation of such a class though? I'm not very familiar with Python's mock